inspirehep/dojson/hep/fields/bd1xx.py

```diff
@@ -24,76 +24,147 @@
 
 from __future__ import absolute_import, division, print_function
 
+import re
+
+from flask import current_app
+from six.moves import zip_longest
+
 from dojson import utils
 
 from ..model import hep, hep2marc
 from ...utils import (
-    create_profile_url,
+    force_single_element,
     get_recid_from_ref,
     get_record_ref,
 )
 
 from inspirehep.utils.helpers import force_force_list
 
 
+ORCID = re.compile('\d{4}-\d{4}-\d{4}-\d{3}[0-9X]')
+
+
 @hep.over('authors', '^[17]00[103_].')
 def authors(self, key, value):
-    """Main Entry-Personal Name."""
-    value = force_force_list(value)
-
-    def get_value(value):
-        affiliations = []
-        if value.get('u'):
-            recid = None
-            try:
-                recid = int(value.get('z'))
-            except:
-                pass
-            affiliations = force_force_list(value.get('u'))
-            record = get_record_ref(recid, 'institutions')
-            affiliations = [{'value': aff, 'record': record} for
-                            aff in affiliations]
-        person_recid = None
-        if value.get('x'):
-            try:
-                person_recid = int(value.get('x'))
-            except:
+    """Authors.
+
+    FIXME: currently not handling 100__v.
+    """
+    def _get_author(value):
+        def _get_affiliations(value):
+            result = []
+
+            u_values = force_force_list(value.get('u'))
+            z_values = force_force_list(value.get('z'))
+
+            if len(u_values) == len(z_values):
+                for u_value, z_value in zip(u_values, z_values):
+                    result.append({
+                        'record': get_record_ref(z_value, 'institutions'),
+                        'value': u_value,
+                    })
+            elif len(u_values) > len(z_values):
+                # TODO: complain to @kaplun
```

Comment:
Err... apparently there were 15k typos in the first 426 records: https://sentry.cern.ch/inspire-sentry/inspire-labs-qa/group/602865/. Now I'm going home, I will check later how bad this number has become and stop the build if needed (I don't want the network people to be angry at me again).